lib/timers.js

```diff
@@ -232,6 +231,12 @@ function listOnTimeout() {
   } else if (list === refedLists[msecs]) {
     delete refedLists[msecs];
```

Comment:
I think that if we check the closing condition weather the list was in a pool or not. 
```diff
+  if (pool[msecs] !== undefined) {
+    this.close();
+  }
```
is better. 
Otherwise, I prefer to check owner property.